### PR TITLE
[exporter/clickhouse] Change the type of `Config.Password` to be `configopaque.String`

### PR DESCRIPTION
**Pull request by Frapschen**
clickhouse exporter also needs use `Config.Password`.

### CODE REVIEW

File: exporter/clickhouseexporter/config_test.go [hunk 1]
```diff
@@ -4,6 +4,7 @@
 package clickhouseexporter
 
 import (
+	"go.opentelemetry.io/collector/config/configopaque"
```
Comment (hanjm) [inline exporter/clickhouseexporter/config_test.go:7]: import group

File: .chloggen/clickhouse-use-cinfigopaue.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (hanjm) [inline .chloggen/clickhouse-use-cinfigopaue.yaml:6]: It is not breaking for user beacuse config value not change, right?

**Review by dmitryax — APPROVED** [review]

File: .chloggen/clickhouse-use-cinfigopaue.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Change the type of `Config.Password` to be `configopaque.String`
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [17273]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (dmitryax) [inline .chloggen/clickhouse-use-cinfigopaue.yaml:20]: Please update the entry to the new interface similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24892/files

It should be a breaking change with `change_logs: [api]`

Comment (Frapschen) [reply .chloggen/clickhouse-use-cinfigopaue.yaml:20]: ok

**Pull request merged**